ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml

```diff
@@ -298,6 +311,7 @@ DappsComboBox {
 
         function onSessionRequest(request) {
             sessionRequestLoader.request = request
+            sessionRequestLoader.requestRejected = false
```

Comment:
WDYT if we leave the `sessionRequestLoader` or the `dappRequestModal` to manage this `requestRejected` flag? I see it as an internal state and makes sense to me to be handled internally.

I think we could reset it in the `rejectRequest`, `Component.onDestruction` or event `sessionRequestLoader.onActiveChange`.

Reply:
That might work. However, the flag is tightly coupled with the current request instance so I like to have this clear in the code and control the order we set it in. Otherwise a `callLater` action might override and change the state.